content/4_Gold/Divisibility.mdx

```diff
@@ -412,6 +412,83 @@ print("\n".join(str(i) for i in ans))
 </PySection>
 </LanguageSection>
 
+### Sum of divisors
+
+<FocusProblem problem = "sumdiv"/>
+
+The sum of all divisors of the integer $N$ can be efficiently computed in $\mathcal{O}(\sqrt{N})$ using [Divisor function](https://en.wikipedia.org/wiki/Divisor_function). The complexity
+comes from the prime factorization of $N$.
+
+The formula states that: Let $d_1^{p_1}, d_2^{p_2}, ..., d_k^{p_k}$ be the prime factorization of $N$, then the divisors' sum is $S = \prod_{i=1}^{k}\frac{d_i^{p_i+1} - 1}{d_i -1}$.
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+ifstream in("sumdiv.in");
+ofstream out("sumdiv.out");
+
+const int MOD = 9901;
+
+int n, m;
+// BeginCodeSnip{Binary Exponentiation}
+int lgpow(int x, int y) {
+	int aux = x, div_sum = 1;
+	for (int bit = 0; (1 << bit) <= y; bit++) {
+		if ((1 << bit) & y) { div_sum = div_sum * aux % MOD; }
+		aux = aux * aux % MOD;
+	}
+	return div_sum;
+}
+// EndCodeSnip
+
+// BeginCodeSnip{Modular Inverse}
+int inv(int x) {
+	if (x <= 1) { return x; }
+	return MOD - MOD / x * inv(MOD % x) % MOD;
+}
+// EndCodeSnip
+
+int main() {
+	in >> n >> m;
+
+	// Prime Factorization
+	long long div_sum = 1;
+	for (int d = 2; d * d <= n; d++) {
+		int p = 0;
+		while (n % d == 0) {
+			n /= d;
+			p++;
+		}
+
+		if (!p) { continue; }
+		p *= m;
+
+		// The Modular Inverse only exists if d % MOD != 0
+		if (d % MOD == 1) {
```

Comment:
weird, might as well do this checking in the inv function

Reply:
Done